spec/features/user_adds_a_new_piece_of_music_spec.rb

```diff
@@ -1,31 +1,50 @@
 require "rails_helper"
 
 feature "users can add a new piece of music" do
-  scenario "user adds new piece of music successfully" do
-    visit new_piece_path
-    expect(page).to have_content "Add New Music"
-    title_and_composer
-    click_button "Add Piece"
 
-    expect(page).to have_content "Piece added successfully"
-  end
+  context "not logged in user" do
+    scenario "does not see link to add a new piece of music" do
+      visit root_path
 
-  scenario "user does not provide proper information" do
-    visit new_piece_path
-    click_button "Add Piece"
-    expect(page).to have_content "Title can't be blank"
+      expect(page).to_not have_content("Add New Piece")
+    end
   end
 
-  scenario "user provides piece and composer pair already added" do
-    visit new_piece_path
-    title_and_composer
-    click_button "Add Piece"
+  context "logged in user" do
+    let(:user) do
+      FactoryGirl.create(:user)
+    end
+
+    before do
+      sign_in_as(user)
+    end
+
+    scenario "user adds new piece of music successfully" do
+      visit new_piece_path
+      expect(page).to have_content "Add New Music"
+      title_and_composer
+      click_button "Add Piece"
+
+      expect(page).to have_content "Piece added successfully"
+    end
+
+    scenario "user does not provide proper information" do
+      visit new_piece_path
+      click_button "Add Piece"
+      expect(page).to have_content "Title can't be blank"
+    end
+
+    scenario "user provides piece and composer pair already added" do
+      visit new_piece_path
+      title_and_composer
+      click_button "Add Piece"
```

Comment:
Instead of lines 38-40, just create a `piece` object, then use that piece object's attributes to fill in the form.